### ISSUE TITLE
Emacs bindings ctrl+j to return and ctrl+i to tab break the use of remap left control to hyper key

### ISSUE THREAD

**Issue by excetara2** [tekezo/Karabiner#653]
I set to use emacs model `ctrl+j` and `ctrl+i` but have hyper key remapped to left control. Then my caps lock key is mapped to left control. The emacs mode keys work perfectly but then the hyper key does not work properly. I can see from the key viewer that if I press the hyper key it pushes down all the keys `ctrl, command, alt, shift` but when `i` is pressed it runs the emacs mode command and then only `command,alt,shift` are pressed. Is there a way to prevent this from happening so these commands only respect ctrl and shift keys because `ctrl+shift+i` is useful too.

**Comment by excetara2** [comment]
So of note is I have the `Fn` key also set to `hyper` and the emacs bindings don't take effect if using the fn key so `fn+j` and `fn+i` will work correctly so can the actual hyper key behave like these.

Or also the commands labeled as (when there are no other modifiers) work correctly with the hyper keys.

Not sure the best way to change this would be implementing something similar to the `Fn` key or to add all the emacs bindings I want with the settings for (when there are no other modifiers).

**Comment by excetara2** [comment]
I found the file where these bindings were set and just remapped them to not use modifiers like some of the other keys to fix this issue in case anyone else has similar issues but I feel maybe these should be the default:

```
    <item>
      <name>Control+I to Tab</name>
      <identifier>option.emacsmode_controlI_myemacs</identifier>
      <not>{{EMACS_MODE_IGNORE_APPS_MINE}}</not>
      <modifier_not>
        ModifierFlag::COMMAND_L, ModifierFlag::COMMAND_R,
        ModifierFlag::OPTION_L,  ModifierFlag::OPTION_R,
      </modifier_not>
      <autogen>__KeyToKey__ KeyCode::I, MODIFIERFLAG_EITHER_LEFT_OR_RIGHT_CONTROL, KeyCode::TAB</autogen>
    </item>
    <item>
      <name>Control+J to Return</name>
      <identifier>remap.controlJ2return_myemacs</identifier>
      <not>{{EMACS_MODE_IGNORE_APPS_MINE}}</not>
      <modifier_not>
        ModifierFlag::COMMAND_L, ModifierFlag::COMMAND_R,
        ModifierFlag::OPTION_L,  ModifierFlag::OPTION_R,
        ModifierFlag::SHIFT_L,   ModifierFlag::SHIFT_R,
      </modifier_not>
      <autogen>__KeyToKey__ KeyCode::J, MODIFIERFLAG_EITHER_LEFT_OR_RIGHT_CONTROL, KeyCode::RETURN</autogen>
    </item>
    <item>
      <name>Control+J to Enter</name>
      <identifier>remap.controlJ2enter_myemacs</identifier>
      <not>{{EMACS_MODE_IGNORE_APPS_MINE}}</not>
      <modifier_not>
        ModifierFlag::COMMAND_L, ModifierFlag::COMMAND_R,
        ModifierFlag::OPTION_L,  ModifierFlag::OPTION_R,
        ModifierFlag::SHIFT_L,   ModifierFlag::SHIFT_R,
      </modifier_not>
      <autogen>__KeyToKey__ KeyCode::J, MODIFIERFLAG_EITHER_LEFT_OR_RIGHT_CONTROL, KeyCode::ENTER</autogen>
    </item>
    <item>
      <name>Control+M to Return</name>
      <identifier>option.emacsmode_controlM_myemacs</identifier>
      <not>{{EMACS_MODE_IGNORE_APPS_MINE}}</not>
      <modifier_not>
        ModifierFlag::COMMAND_L, ModifierFlag::COMMAND_R,
        ModifierFlag::OPTION_L,  ModifierFlag::OPTION_R,
        ModifierFlag::SHIFT_L,   ModifierFlag::SHIFT_R,
      </modifier_not>
      <autogen>__KeyToKey__ KeyCode::M, MODIFIERFLAG_EITHER_LEFT_OR_RIGHT_CONTROL, KeyCode::RETURN</autogen>
    </item>

```

**Comment by tekezo** [comment]
Your workaround is a correct way.

I'm loving this behavior (eg. press command-control-I to send command-Tab).
So I cannot determine we should change the default behavior.

But I've added the strict version.
Please use these version in the latest beta version or continue using your private.xml version.
https://github.com/tekezo/Karabiner/commit/58319846b13683c5f0636679de5b343f496bd10c


How to upgrade to the latest beta version:
https://pqrs.org/osx/karabiner/document.html#check-for-updates

**Comment by excetara2** [comment]
Actually I never thought of that so I've actually removed the command from my version but left the alt in place as I don't see why it would be needed. I already had removed shift because that is really useful even with without command+tab but it makes sense to use with that too. One app I wanted this behavior except for shift not working with control or super so I've just made the change to that app and all is good.
